subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -57,7 +59,7 @@
  * gradle someJavaExecTask --debug-jvm
  * </pre>
  */
-public class JavaExec extends ConventionTask implements JavaExecSpec {
+public class JavaExec extends ConventionTask implements JavaExecSpec, Cancellable {
```

Comment:
❌ I don't think we should do this at the task level, but instead make *every* exec action (no matter if through a task or invoked via `Project.exec`) cancellable.

Reply:
@ghale I'd be interested in your take on this - Would there be a reason to only do this for some exec actions or is a transparent cancellation of all of them reasonable?

Reply:
I think we really have two different goals here.  One is that we want exec actions to be cancellable (wherever they occur) and two is that we want any arbitrary task to be cancellable.  I think the former covers tasks that exec things, and the latter covers exec actions that are a part of tasks.  Realistically, I don't think there is an order here - we can do either without necessarily doing the other.  But, per your other comments, we should split the story up into parts.  I also like the idea of a more generic cancellation api rather than just cancelling tasks.